sdv/constraints/tabular.py

```diff
@@ -211,78 +214,109 @@ class GreaterThan(Constraint):
-        if self._low_is_scalar:
-            return self._low
-        elif self._low in table_data.columns:
-            return table_data[self._low]
+    _columns_to_reconstruct = None
 
-        return None
+    @staticmethod
+    def _validate_scalar(scalar_column, column_names, scalar):
+        """Validate scalar comparison inputs.
 
-    def _get_high_value(self, table_data):
-        if self._high_is_scalar:
-            return self._high
-        elif self._high in table_data.columns:
-            return table_data[self._high]
+        - Make sure that the scalar column is not a list and raise the proper error if it is.
+        - If the `column_names` is not a list it would make it a list.
+        - Return both the scalar column and column names with the right format
+        """
+        if isinstance(scalar_column, list):
+            raise TypeError(f'{scalar} cannot be a list when scalar="{scalar}".')
+
+        column_names = [column_names] if not isinstance(column_names, list) else column_names
+        return column_names
+
+    @classmethod
+    def _validate_inputs(cls, low, high, scalar):
+        if scalar is None:
+            low = [low] if not isinstance(low, list) else low
+            high = [high] if not isinstance(high, list) else high
+            if len(low) > 1 and len(high) > 1:
+                raise ValueError('either `high` or `low` must contain only one column.')
+
+            constraint_columns = tuple(low + high)
+
+        elif scalar == 'low':
+            high = cls._validate_scalar(scalar_column=low, column_names=high, scalar=scalar)
+            constraint_columns = tuple(high)
+        elif scalar == 'high':
+            low = cls._validate_scalar(scalar_column=high, column_names=low, scalar=scalar)
+            constraint_columns = tuple(low)
+        else:
+            raise ValueError(f"scalar {scalar} is unknown. Use either: 'high', 'low', or None.")
 
-        return None
+        return low, high, constraint_columns
 
-    def _get_column_to_reconstruct(self):
+    def _get_columns_to_reconstruct(self):
         if self._drop == 'high':
             column = self._high
         elif self._drop == 'low':
             column = self._low
-        elif self._high_is_scalar:
+        elif self._scalar == 'high':
             column = self._low
         else:
             column = self._high
 
         return column
 
-    def _get_diff_column_name(self, table_data):
-        token = '#'
-        if len(self.constraint_columns) == 1:
-            name = self.constraint_columns[0] + token
-            while name in table_data.columns:
-                name += '#'
+    def __init__(self, low, high, strict=False, handling_strategy='transform',
+                 fit_columns_model=True, drop=None, scalar=None):
+        self._strict = strict
+        self._drop = drop
+        self._scalar = scalar
+        self._low, self._high, self.constraint_columns = self._validate_inputs(low, high, scalar)
```

Comment:
I just realized about one thing: We should be validating `drop` too, since `scalar == drop` is no possible.

Reply:
Yes, I believe `low = low` and `high = high` is not possible; but `None` and `None` works.. I will add one more check to `_validate_inputs`.